test/Constraints/diagnostics.swift

```diff
@@ -672,7 +672,7 @@ func overloadSetResultType(_ a : Int, b : Int) -> Int {
   // https://twitter.com/_jlfischer/status/712337382175952896
   // TODO: <rdar://problem/27391581> QoI: Nonsensical "binary operator '&&' cannot be applied to two 'Bool' operands"
   return a == b && 1 == 2  // expected-error {{binary operator '&&' cannot be applied to two 'Bool' operands}}
-  // expected-note @-1 {{expected an argument list of type '(Bool, @autoclosure () throws -> Bool)'}}
+  // expected-note @-1 {{expected an argument list of type '(Bool, () throws -> Bool)'}}
```

Comment:
Also, the code path might be a little different for operators. I'll look into this, and otherwise also expand test coverage for operators with e.g. escaping